src/renderer/services/neo.js

```diff
@@ -326,17 +327,33 @@ export default {
       try {
         const inMemory = _.get(store.state.transactionDetails, hash);
         if (inMemory) {
-          inMemory.currentBlockHeight = network.getSelectedNetwork().bestBlock.index;
-          inMemory.confirmations = inMemory.currentBlockHeight - inMemory.block;
+          if (network.getSelectedNetwork().bestBlock) {
+            inMemory.currentBlockHeight = network.getSelectedNetwork().bestBlock.index;
+            inMemory.confirmations = inMemory.currentBlockHeight - inMemory.block;
+          }
           return resolve(inMemory);
         }
 
         return rpcClient.getRawTransaction(hash, 1)
           .then((transaction) => {
-            transaction.currentBlockHeight = network.getSelectedNetwork().bestBlock.index;
+            const inputPromises = [];
+
             if (transaction.confirmations > 0) {
               transaction.confirmed = true;
-              transaction.block = transaction.currentBlockHeight - transaction.confirmations;
+
+              // Look up the block from the blockhash
+              inputPromises.push(new Promise((resolve, reject) => {
+                store.dispatch('fetchBlockHeaderByHash', {
+                  blockHash: transaction.blockhash,
+                  done: ((data) => {
+                    resolve(data);
+                  }),
+                  failed: ((ex) => { reject(ex); }),
```

Comment:
Not a big deal, but why `ex`?

Reply:
ex short for exception

Reply:
This is the only time its ok to use a single-letter variable (`e`). Its also pretty common across our app and JS in general.